File: week-5/find-bottom-left-tree-value/find-bottom-left-tree-value.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def findBottomLeftValue(self, root: TreeNode) -> int:
        if not root: return None
        
        queue = collections.deque([root])
        while queue:
            size = len(queue)
            arr = []
            while size:
                cur = queue.popleft()
                arr.append(cur.val)
                size -= 1
                if cur.left:
                    queue.append(cur.left)
                if cur.right:
                    queue.append(cur.right)
        
        return arr[0]

File: week-5/find-bottom-left-tree-value/find-bottom-left-tree-value.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var findBottomLeftValue = function(root) {
    if (!root) return null;
    
    queue = [root];
    while (queue.length) {
        size = queue.length;
        arr = [];
        while (size) {
            cur = queue.shift();
            arr.push(cur.val);
            if (cur.left) queue.push(cur.left);
            if (cur.right) queue.push(cur.right);
            --size;
        }
    }
    
    return arr[0];
};
